RecoEgamma/EgammaElectronProducers/plugins/alpaka/PixelMatchingAlgo.dev.cc

```diff
@@ -0,0 +1,308 @@
+// Check that ALPAKA_HOST_ONLY is not defined during device compilation:
+#ifdef ALPAKA_HOST_ONLY
+#error ALPAKA_HOST_ONLY defined in device compilation
+#endif
+
+#include <alpaka/alpaka.hpp>
+
+#include "DataFormats/EgammaReco/interface/alpaka/EleSeedDeviceCollection.h"
+#include "DataFormats/EgammaReco/interface/alpaka/SuperclusterDeviceCollection.h"
+#include "MagneticField/ParametrizedEngine/interface/ParabolicParametrizedMagneticField.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixBarrelPlaneCrossingByCircle.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/ftsFromVertexToPointPortable.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixArbitraryPlaneCrossing.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixForwardPlaneCrossing.h"
+
+#include "PixelMatchingAlgo.h"
+#include "DataFormats/EgammaReco/interface/alpaka/EleRelPointPairPortable.h"
+#include "DataFormats/EgammaReco/interface/alpaka/Plane.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+        template<typename TAcc, typename T>
+	ALPAKA_FN_ACC ALPAKA_FN_INLINE T getZVtxFromExtrapolation(TAcc const& acc,
+								  const Vec3d& primeVtxPos,
+								  const Vec3d& hitPos,
+								  const Vec3d& candPos) {
+		auto sq = [](T x) { return x * x; };
+
+		auto calRDiff2 = [sq](const Vec3d& p1, const Vec3d& p2) {
+		  return sq(p2[0] - p1[0]) + sq(p2[1] - p1[1]);
+		};
+		const T r1Diff = alpaka::math::sqrt(acc, calRDiff2(primeVtxPos, hitPos));
+		const T r2Diff = alpaka::math::sqrt(acc, calRDiff2(hitPos, candPos));
+
+		T zvtx = hitPos[2] - r1Diff * (candPos[2] - hitPos[2]) / r2Diff;
+		return zvtx;
+	}
+
+	template<typename TAcc, typename T>
+	ALPAKA_FN_ACC ALPAKA_FN_INLINE T getCutValue(TAcc const& acc, const T et, const T highEt, const T highEtThres, const T lowEtGrad) {
+		return highEt + alpaka::math::min(acc, static_cast<T>(0.), et - highEtThres) * lowEtGrad;
+	}
+
+	//--- Kernel for printing the SC SoA
+	class printSCSoA {
+	public:
+	template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+	ALPAKA_FN_ACC void operator()(TAcc const& acc,
+									reco::SuperclusterDeviceCollection::View view,
```

Comment:
```suggestion
									reco::SuperclusterDeviceCollection::ConstView view,
```
because the `view is not modified